src/resolve/metadataResolver.ts

```diff
@@ -174,8 +174,8 @@ export class MetadataResolver {
           // any of the following 3 options is considered a good match
           // mixedContent and bundles don't have a suffix to match
           ['mixedContent', 'bundle'].includes(type.strategies?.adapter) ||
-          // the suffix matches the type we think it is
-          (type.suffix && fsPath.endsWith(`${type.suffix}${META_XML_SUFFIX}`)) ||
+          // the file suffix (in source or mdapi format) matches the type suffix we think it is
+          (type.suffix && [type.suffix, `${type.suffix}${META_XML_SUFFIX}`].some((s) => fsPath.endsWith(s))) ||
```

Comment:
that exception is that little punk emailServicesFunction for whom `.xml` is the suffix.

now that we're testing with a real registry, can you make sure that it's handled correctly?

Reply:
FYI, I was wanting it in a strictDir.  ESF has a habit of matching way too promiscuously and I'm trying to keep that from happening.

https://github.com/forcedotcom/source-deploy-retrieve/pull/595/files#diff-af7a780a5aa66c4831707c3ccd4e24d4f88a6639346bc0b1897c287efcbe14c9

So...what would happen if it *were* in strictDir?

Reply:
Testing would help us out, but I think it would still work just fine because it would only get to the suffix matching if the path contained the expected directory.

Reply:
After testing, ESF still retrieves and deploys successfully when defined as a strictDirectoryName.